node/src/neighborhood/mod.rs

```diff
@@ -1245,13 +1392,55 @@ impl Neighborhood {
         return_route_id
     }
 
+    pub fn find_exit_location<'a>(
+        &'a self,
+        source: &'a PublicKey,
+        minimum_hops: usize,
+        payload_size: usize,
+    ) -> Vec<&'a PublicKey> {
+        let mut minimum_undesirability = i64::MAX;
+        let initial_undesirability = 0;
+        let research_exits: &mut Vec<&'a PublicKey> = &mut vec![];
+        let mut prefix = Vec::with_capacity(10);
```

Comment:
Tohle je zajimava otazka, jestli to mnozstvi predalokovane pameti je dostatecne nebo ne. Bylo by zajimave experimentovat s touto hodnotou (zmensit ji nebo zvetsit) a sledovat zmeny v delce trvani toho celkoveho vypoctu. Aby ta hodnota nebyla az tak moc ad hoc. Pokud to nechces delat, muzes to treba jen sminit v komentari s TODO.

Jinak Utkarsh by po tobe chtel, abys k te hodnote priradil konstantu a pouzil nasledne ji. V podstate to delame se vsemi takovyhlemi hodnotami, abychom dali jasne najevo, ze je to podstatny parameter, ktery byl zvoleny uvazlive. Nejlepe kdyz te konstante das nejak smysluplne jmeno (napr. MULTIPLE_OF_PREALOCATED_MEMORY nebo neco podobneho).